src/main/java/kostka/moviecatalog/service/UserService.java

```diff
@@ -0,0 +1,85 @@
+package kostka.moviecatalog.service;
+
+import kostka.moviecatalog.dto.UserDto;
+import kostka.moviecatalog.entity.Role;
+import kostka.moviecatalog.entity.User;
+import kostka.moviecatalog.exception.FutureBirthDateException;
+import kostka.moviecatalog.exception.UserNameNotUniqueException;
+import kostka.moviecatalog.repository.RoleRepository;
+import kostka.moviecatalog.repository.UserRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.Set;
+
+import static kostka.moviecatalog.configuration.WebSecurityConfiguration.USER_ROLE;
+
+@Service
+public class UserService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserService.class);
+    private UserRepository userRepository;
+    private RoleRepository roleRepository;
+
+    BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
```

Comment:
Was not this specified as a bean somewhere? I'd better use that.

Reply:
true, thanks